### ISSUE TITLE
File whitelisting for npm publish

### ISSUE THREAD

**Issue by springmeyer** [mapbox/hpp-skel#10]
tl;dr let's standardize on https://docs.npmjs.com/files/package.json#files

## background

The use of npm publishing is important for hpp-skel because header-only C++ libs are frequently used in node.js C++ addons. With npm publishing we want to ensure only the minimal files are included in the published tarball. The normal approach for this is to depend on `.npmignore`. `.npmignore` works the same as `.gitingore`: you `blacklist` things you don't want packaged (and by default if no `.npmingore` is used then `.gitignore` will be used in its place.

## problem
But this solution does not work for a `skel` repo. Here we can't know what files will be created by authors using `skel` as a starting point. So, blacklisting is full of fail.

## solution
So, we want to use whitelisting instead. The convention for header-only libs based on `hpp-skel` is that only the `include_dirs.js` and the `include/` directory (in addition to the automatically included `README.md` and `packag.json`) will need to be packaged. Because there is no support for `.npminclude` (https://github.com/npm/npm/issues/13171) we can do this with the package.json `files` feature: https://docs.npmjs.com/files/package.json#files
 
/cc @mapsam @GretaCB @mikemorris

**Comment by springmeyer** [comment]
Note: to get this working we'll need to remove this empty `.npmignore`: https://github.com/mapbox/hpp-skel/blob/master/.npmignore

**Comment by mapsam** [comment]
Hey @springmeyer picking this up again - should we include the Makefile as well?

**Comment by springmeyer** [comment]
We are no longer recommending publishing to npm. Rather we will only publish to mason. This issue has some history on the switch in mindset: https://github.com/mapbox/cpp/issues/20. Only using `mason` avoids having to publish to two places. This also means that, to install headers on windows, we'll need a small mason client written in javascript (https://github.com/mapbox/mason/issues/396). @mapsam - this would probably be a 1-2 day project to scope out - let's chat about this voice.

**Comment by springmeyer** [comment]
To be clear: so this issue can be closed. And we no longer need a `package.json` in this repo.

**Comment by mapsam** [comment]
Sweet! I figured we were nearing that case. I'm very 👍 avoiding NPM entirely here.
